source/install/config-cluster-transport-encryption.rst

```diff
@@ -0,0 +1,220 @@
+=============================================
+Configuring Cluster Transport Encryption (E20)
+=============================================
+
+Mattermost is able to encrypt the messages sent within the cluster of a deployment
+using SSH tunneling. The guide walks through the deployment of this solution on
+Ubuntu 18.04, but it can be adapted for any Linux operating system.
+
+While this document only describes the configuration of a 3 node cluster, it is
+by no means limited to that amount.
+
+**Prerequisites:**
+- SSH port whitelisted between each node of the deployment
+- ufw / iptables is active on each node
+- Access to the root / sudo user of each node for configuration
+- A configured Mattermost cluster
+- Mattermost running with a dedicated service user
+- Mattermost service is stopped on each cluster node
+
+**Note:** A support on the application level is currently in development and will
+deprecate this document.
+
+Example Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In this scenario there are 3 application nodes in our environment with the following
+hostname / IP mapping:
+
+- **transport-encryption-mattermost1:** 10.10.250.146
+- **transport-encryption-mattermost2:** 10.10.250.231
+- **transport-encryption-mattermost3:** 10.10.250.165
+
+Preparations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Connect to each Mattermost server with a sudo or root user
+- Make a note of the IP from each cluster member used for the internal communication
+- Ensure `AllowTcpForwarding` is enabled in `/etc/ssh/sshd_config` of each cluster node
+
+SSH Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+On each node generate a SSH key-pair for the service account, in our scenario
+called `mattermost`:
+
+```
```

Comment:
Please see https://docs.mattermost.com/process/documentation-guidelines.html#literal-blocks for how to add code blocks.